dbfit-java/core/src/main/java/dbfit/api/AbstractDbEnvironment.java

```diff
@@ -273,5 +273,9 @@ private static boolean isConnected(final Connection conn) throws SQLException {
         return (conn != null && !conn.isClosed());
     }
 
+    @Override
+    public String getActualErrorCode(SQLException e) {
```

Comment:
Some wondering what `actual` stands for, what's the difference between `error code` and `actual error code`?

At first glimpse it looks like SQLException is sometimes kind of wrapper exception without details about the underlying root cause so here we try to get this more detailed underlying error, right?

Reply:
> At first glimpse it looks like SQLException is sometimes kind of wrapper exception without details about the underlying root cause so here we try to get this more detailed underlying error, right?

Well, yes, it's partly that and partly so that we can provide abstraction from the need to call either `e.getErrorCode` or `e.getSQLState` depending upon the DB type.